### DISCUSSION TITLE
How to reset password of Phone Auth User ?

### DISCUSSION THREAD

**Discussion by dukesx** [supabase/supabase#4712] [Questions] [answered]
Hi everyone,

I am using message bird to login with a password and phone number but i am neither able to add an email to the logged in user nor am i able to reset password for such a user. Anybody can help in this regard ?

**Comment by J0** [comment, marked as answer]
Hey,

Thanks for your query -- you can make use of the [update endpoints](https://supabase.com/docs/reference/javascript/auth-update) on the server or client (depending on your usecase) in order to update the password for a user.

**Reply by matart15** [reply]
@J0 this only works if user is already authenticated.

How do i implement Forgot password on mobile users?
I tried [resetPasswordForEmail](https://supabase.com/docs/reference/javascript/auth-resetpasswordforemail)

but it gives me url which returns `{"code": "12dd191b-cac0-4fa1-9bd1-3072
b3c6e0f7"}`

But there is no info on how to use this code
